mods/taskbar-icon-numberer.wh.cpp

```diff
@@ -0,0 +1,468 @@
+        
+        // Collect all TaskListButtons with their X positions
+        std::vector<std::pair<double, FrameworkElement>> buttons;
+        
+        for (uint32_t i = 0; i < children.Size(); i++) {
+            auto child = children.GetAt(i).try_as<FrameworkElement>();
+            if (!child || child.Name() != L"TaskListButton") continue;
+            
+            double x = child.ActualOffset().x;
+            buttons.push_back({x, child});
+        }
+        
+        // Sort by X position (left to right)
+        std::sort(buttons.begin(), buttons.end(), 
+                  [](const auto& a, const auto& b) { return a.first < b.first; });
+        
+        // Number buttons in visual order
+        int buttonIndex = 0;
+        for (const auto& [x, button] : buttons) {
+            if (buttonIndex >= g_settings.maxNumbers) break;
+            
+            bool shouldShow = true;
+            if (g_settings.showOnlyRunning) {
+                shouldShow = TaskListButton_IsRunning(button);
+            }
+            
+            if (shouldShow) {
+                buttonIndex++;
+                CreateNumberOverlay(button, buttonIndex);
+            } else {
+                // Remove overlay
+                auto iconPanel = FindChildByName(button, L"IconPanel");
+                if (iconPanel) {
+                    auto existingOverlay = FindChildByName(iconPanel, L"WindhawkNumberOverlay");
+                    if (existingOverlay) {
+                        auto panelChildren = iconPanel.as<Panel>().Children();
+                        uint32_t index;
+                        if (panelChildren.IndexOf(existingOverlay, index)) {
+                            panelChildren.RemoveAt(index);
+                        }
+                    }
+                }
+            }
+        }
+        
+    } catch (...) {
+        // Ignore errors
+    }
+}
+
+void WINAPI TaskListButton_UpdateVisualStates_Hook(void* pThis) {
+    if (TaskListButton_UpdateVisualStates_Original) {
+        TaskListButton_UpdateVisualStates_Original(pThis);
+    }
+    
+    if (g_unloading) return;
+    
+    try {
+        void* taskListButtonIUnknownPtr = (void**)pThis + 3;
+        winrt::Windows::Foundation::IUnknown taskListButtonIUnknown;
+        winrt::copy_from_abi(taskListButtonIUnknown, taskListButtonIUnknownPtr);
+
+        auto taskListButtonElement = taskListButtonIUnknown.as<FrameworkElement>();
+        if (!taskListButtonElement) return;
+
+        // Find parent repeater
+        auto parent = Media::VisualTreeHelper::GetParent(taskListButtonElement).as<FrameworkElement>();
+        if (!parent || parent.Name() != L"TaskbarFrameRepeater") return;
+
+        // Update all buttons at once to ensure consistent numbering
+        UpdateAllTaskbarNumbers(parent);
+        
+    } catch (...) {
+        // Ignore errors
+    }
+}
+
+bool HookTaskbarViewDllSymbols(HMODULE module) {
+    WindhawkUtils::SYMBOL_HOOK user32DllHooks[] = {
```

Comment:
The variable name is incorrect. A correct example for this case:
https://github.com/ramensoftware/windhawk-mods/blob/4f32b2163d3b8938660000bb6cf62634e022635c/mods/taskbar-on-top.wh.cpp#L1789-L1790

Reply:
Add the comment too.